Source/Core/Core/HW/GCMemcardDirectory.cpp

```diff
@@ -186,10 +188,54 @@ GCMemcardDirectory::GCMemcardDirectory(const std::string& directory, int slot, u
 	m_dir1.fixChecksums();
 	m_dir2 = m_dir1;
 	m_bat2 = m_bat1;
+
+	m_flush_thread = std::thread(&GCMemcardDirectory::FlushThread, this);
+}
+
+void GCMemcardDirectory::FlushThread()
+{
+	if (!SConfig::GetInstance().m_LocalCoreStartupParameter.bEnableMemcardSaving)
+	{
+		return;
+	}
+
+
+	Common::SetCurrentThreadName(
+		StringFromFormat("Memcard%x-Flush", card_index).c_str());
+
+	while (true)
+	{
+		// no-op until signalled
+		m_flush_trigger.Wait();
+
+		if (m_exiting)
+		{
+			m_exiting = false;
+			return;
+		}
+		// no-op as long as signalled within flush_interval
```

Comment:
There's always the flush in the destructor.
I didn't put too much work in the algorithm, really just aimed at not deadlocking, and low disk io.

That was my intention, not a very sophisticated method, but minimizes disk io (maybe unnecessary because of OS level caching).
blindly signal after each page, rather then test for the end of a block, although maybe branch prediction would be better, really would need to benchmark.